### PR TITLE
Paint Road

### PR DESCRIPTION
**Pull request by duncanspumpkin**
~~WiP. I've got the tram code only atm.~~ ~~mostly works apart from street lights~~

Works

### CODE REVIEW

File: src/OpenLoco/src/Paint/PaintRoadStyle0Data.h [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+#include <array>
+#include <span>
+
+namespace OpenLoco::Paint::Style0
+{
+    // 0x004083B1, 0x004084BC, 0x004083B1, 0x004084BC
+    constexpr RoadPaintMergeablePiece kStraight0 = {
+        std::array<uint32_t, 4>{ 3486, 3487, 3486, 3487 },
+        /* StreetlightHeights */ kNoStreetlights,
+        /* IsMultiTileMerge */ true,
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Paint/PaintRoadStyle0Data.h:15]: Given we've been using C++20 for a while, can't we just use designated initialisers?

Comment (duncanspumpkin) [reply src/OpenLoco/src/Paint/PaintRoadStyle0Data.h]: Yes we should do that.

Comment (duncanspumpkin) [reply src/OpenLoco/src/Paint/PaintRoadStyle0Data.h]: No can't do that due to using a constructor

**Comment by duncanspumpkin** [comment]
This should now be ready for testing/review. I noticed that vanilla was using different bounding box offsets for streetlight when they are on flat single tiles compared to others. I'm not sure why though. I've decided not to do that.

**Review by AaronVanGeffen — APPROVED** [review]
After the initial smoke test passed, I took giant screenshots of the four rotations on this branch and master. An image comparison tool reveals **no** differences! Nice!
```
$ odiff scr_master/1.png scr_paintroad/1.png diff1.png
Success! Images are equal.
No diff output created.
$ odiff scr_master/2.png scr_paintroad/2.png diff2.png
Success! Images are equal.
No diff output created.
$ odiff scr_master/3.png scr_paintroad/3.png diff3.png
Success! Images are equal.
No diff output created.
$ odiff scr_master/4.png scr_paintroad/4.png diff4.png
Success! Images are equal.
No diff output created.
```

**Pull request merged**